Tools/CMake/torque3d.cmake

```diff
@@ -645,11 +645,13 @@ else()
   endif()
 endif()
 
-if(NOT EXISTS "${projectOutDir}/${PROJECT_NAME}.torsion")
-    CONFIGURE_FILE("${cmakeDir}/template.torsion.in" "${projectOutDir}/${PROJECT_NAME}.torsion")
-endif()
-if(EXISTS "${CMAKE_SOURCE_DIR}/Templates/${TORQUE_TEMPLATE}/game/main.${TORQUE_SCRIPT_EXTENSION}.in")
-    CONFIGURE_FILE("${CMAKE_SOURCE_DIR}/Templates/${TORQUE_TEMPLATE}/game/main.${TORQUE_SCRIPT_EXTENSION}.in" "${projectOutDir}/main.${TORQUE_SCRIPT_EXTENSION}")
+CONFIGURE_FILE("${cmakeDir}/template.torsion.in" "${projectOutDir}/${PROJECT_NAME}.torsion")
+if(EXISTS "${CMAKE_SOURCE_DIR}/Templates/${TORQUE_TEMPLATE}/game/main.tscript.in")
+    CONFIGURE_FILE("${CMAKE_SOURCE_DIR}/Templates/${TORQUE_TEMPLATE}/game/main.tscript.in" "${projectOutDir}/main.${TORQUE_SCRIPT_EXTENSION}")
+else()
+    if(EXISTS "${CMAKE_SOURCE_DIR}/Templates/${TORQUE_TEMPLATE}/game/main.${TORQUE_SCRIPT_EXTENSION}.in")
```

Comment:
I would prefer swapping these if-statements so main.cs.in has precedence over main.tscript.in if you are targeting main.cs
But I don't really see the use-case here tbh

Reply:
The way it was originally written would only configure a main._scriptExension_ file if the template contained the script extension that was configured. Since the stock templates will all contain .tscript.in, I made that the first case and left the original as secondary. The only use case for the second is older templates that still contain main.cs.in.

Reply:
Ah that makes sense! :)